tests/others/test_attention_backends.py

```diff
@@ -0,0 +1,264 @@
+"""
+This test suite exists for the maintainers currently. It's not run in our CI at the moment.
+
+Once attention backends become more mature, we can consider including this in our CI.
+
+To run this test suite:
+
+```bash
+export RUN_ATTENTION_BACKEND_TESTS=yes
+export DIFFUSERS_ENABLE_HUB_KERNELS=yes
+
+pytest tests/others/test_attention_backends.py
+```
+
+Tests were conducted on an H100 with PyTorch 2.8.0 (CUDA 12.9). Slices for the compilation tests in
+"native" variants were obtained with a torch nightly version (2.10.0.dev20250924+cu128).
+"""
+
+import os
+
+import pytest
+import torch
+
+
+pytestmark = pytest.mark.skipif(
+    os.getenv("RUN_ATTENTION_BACKEND_TESTS", "false") == "false", reason="Feature not mature enough."
+)
+from diffusers import FluxPipeline  # noqa: E402
+from diffusers.utils import is_torch_version  # noqa: E402
+
+
+FORWARD_CASES = [
+    ("flash_hub", None),
+    (
+        "_flash_3_hub",
+        torch.tensor(
+            [
+                0.0820,
+                0.0859,
+                0.0938,
+                0.1016,
+                0.0977,
+                0.0996,
+                0.1016,
+                0.1016,
+                0.2188,
+                0.2246,
+                0.2344,
+                0.2480,
+                0.2539,
+                0.2480,
+                0.2441,
+                0.2715,
+            ],
+            dtype=torch.bfloat16,
+        ),
+    ),
+    (
+        "native",
+        torch.tensor(
+            [
+                0.0820,
+                0.0859,
+                0.0938,
+                0.1016,
+                0.0957,
+                0.0996,
+                0.0996,
+                0.1016,
+                0.2188,
+                0.2266,
+                0.2363,
+                0.2500,
+                0.2539,
+                0.2480,
+                0.2461,
+                0.2734,
+            ],
+            dtype=torch.bfloat16,
+        ),
+    ),
+    (
+        "_native_cudnn",
+        torch.tensor(
+            [
+                0.0781,
+                0.0840,
+                0.0879,
+                0.0957,
+                0.0898,
+                0.0957,
+                0.0957,
+                0.0977,
+                0.2168,
+                0.2246,
+                0.2324,
+                0.2500,
+                0.2539,
+                0.2480,
+                0.2441,
+                0.2695,
+            ],
+            dtype=torch.bfloat16,
+        ),
+    ),
+]
+
+COMPILE_CASES = [
+    ("flash_hub", None, True),
```

Comment:
Will add add the test slices after https://github.com/huggingface/diffusers/pull/12387 is merged.